### ISSUE TITLE
zsh completion instructions incorrect, maybe

### ISSUE THREAD

**Issue by xxxserxxx** [martinvonz/jj#1393]
I think maybe the README instructions didn't get updated for a recent change -- or else the completion output changed unexpectedly.

The README says to strip the last line of `jj support completion --zsh` with `sed`, but that now removes the closing `fi` of the last if statement:

```bash
$ jj support completion --zsh | tail -n 5
if [ "$funcstack[1]" = "_jj" ]; then
    _jj "$@"
else
    compdef _jj jj
fi
```

**Comment by martinvonz** [comment]
Yes, it look like that changed in https://github.com/clap-rs/clap/pull/4734/commits/0f3e729358ff2bc348ef54e1307d27722cfe0311, which was released in `clap-complete` 4.1.4, so you won't technically run into that problem if you follow our installation instructions and pass `--locked` to `cargo install` since [we depend on 4.0.7](https://github.com/martinvonz/jj/blob/e2b4d7058daa4ea6efb8b424387fcf6040c73a0a/Cargo.toml#L39). I'll leave this bug open until we have upgraded to the new version of `clap-complete` and updated the documentation.

FYI, this seems to work for me with the new version:
```shell
autoload -U compinit
compinit
source <(jj support completion --zsh)
```

**Comment by xxxserxxx** [comment]
Yup! The completion seemed fine, it was just the README instruction was missed. Thanks
